dace/codegen/targets/rtl.py

```diff
@@ -213,196 +225,218 @@ def generate_rtl_inputs_outputs(self, buses, scalars):
+                        outputs[name] = f'{name}[out_ptr_{name}++] = (int)(model->m_axis_{name}_tdata);'
+                else:
+                    outputs[name] = f'{name}[out_ptr_{name}++] = (int)(model->m_axis_{name}_tdata);'
+
+            else:  # input
+                conn = tasklet.in_connectors[name]
+                if isinstance(conn, dtypes.vector):
+                    inputs[name] = f'''in_ptr_{name}++;
+for (int j = 0; j < {veclen}; j++) {{
+    model->s_axis_{name}_tdata[j] = {name}[j];
+}}'''
+                elif isinstance(conn, dtypes.pointer):
+                    if isinstance(conn.base_type, dtypes.vector):
+                        inputs[name] = f'''int idx = in_ptr_{name}++;
+for (int j = 0; j < {veclen}; j++) {{
+    model->s_axis_{name}_tdata[j] = (int){name}[idx][j];
+}}'''
+                    else:
+                        inputs[name] = f'model->s_axis_{name}_tdata = {name}[in_ptr_{name}++];'
+                else:
+                    inputs[name] = f'''in_ptr_{name}++;
+model->s_axis_{name}_tdata = {name}[0];'''
+
+        return inputs, outputs
 
     def generate_cpp_vector_init(self, tasklet):
-        init_vector_string = "\n".join([
-            """\
-        for(int i = 0; i < {veclen}; i++){{
-         model->{name}[i] = 0;
-        }}\
-        """.format(veclen=tasklet.in_connectors[var_name].veclen, name=var_name) if isinstance(
-                tasklet.in_connectors[var_name], dtypes.vector) else "" for var_name in tasklet.in_connectors
-        ])
-        return "// initialize vector\n" if len(init_vector_string) > 0 else "" + init_vector_string
-
-    def generate_cpp_num_elements(self, tasklet):
+        inits = []
+        for name in tasklet.in_connectors:
+            conn = tasklet.in_connectors[name]
+            if isinstance(conn, dtypes.pointer):
+                conn = conn.base_type
+            if isinstance(conn, dtypes.vector):
+                inits.append(f'''for (int j = 0; j < {conn.veclen}; j++) {{
+    model->s_axis_{name}_tdata[j] = 0;
+}}''')
+
+        return "\n".join(inits)
+
+    def generate_cpp_num_elements(self, buses):
         # TODO: compute num_elements=#elements that enter/leave the pipeline, for now we assume in_elem=out_elem (i.e. no reduction)
-        ins = [f'int num_elements_{name} = 1;' for name in tasklet.in_connectors]
-        outs = [f'int num_elements_{name} = 1;' for name in tasklet.out_connectors]
-        return ins + outs
+        return [
+            f'''int num_elements_{name} = {volume};'''
+            for name, (arr, is_output, bytes, veclen, volume) in buses.items()
+        ]
 
     def generate_cpp_internal_state(self, tasklet):
         internal_state_str = " ".join(
             ["{}=0x%x".format(var_name) for var_name in {
                 **tasklet.in_connectors,
                 **tasklet.out_connectors
             }])
-        internal_state_var = ", ".join(
+        internal_state_var = ", ".join(  # TODO fix
```

Comment:
Fix what? ;-)

Reply:
Fix the debug statements :) this should have been fixed now.